### PR TITLE
[navigation]Refactor: flatten left nav in Analytics(all) use case

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description

As the multi levels of left nav may lead to context missing and confusion, we will optimize the analytics(all) use case nav by flattening all the navigation. And in order to keep backward compatible, we will keep using `showInAllNavGroup` for those links within nav group category. 

#### Dependencies

- [x] https://github.com/opensearch-project/dashboards-observability/pull/2182
- [x] https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/883
- [x] https://github.com/opensearch-project/dashboards-maps/pull/674
- [x] https://github.com/opensearch-project/dashboards-reporting/pull/448
- [x] https://github.com/opensearch-project/security-analytics-dashboards-plugin/pull/1169
- [x] https://github.com/opensearch-project/alerting-dashboards-plugin/pull/1114

### Analytics(all)
image

### Observability
image

### Security analytics
image

### Search
image

### Essentials
image

## Changelog

- feat: [navigation] flatten left nav in Analytics(all) use case
- feat: [navigation] Adjust the appearances of the left navigation menu and the landing page

### Check List

- [x] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/core/public/chrome/nav_group/nav_group_service.ts
Comment (Hailong-am) [inline src/core/public/chrome/nav_group/nav_group_service.ts:186]: for those links that will exist in multiple use cases like visualization, will navLinksResult have duplicate records?

**Review by Hailong-am — COMMENTED** [review]
do we need to update here to make sure nav group will not change when in all use case/ 
``` ts   
this.currentAppIdSubscription = combineLatest([
      application.currentAppId$,
      this.getSortedNavGroupsMap$(),
    ]).subscribe(([appId, navGroupMap]) => {
```

File: src/core/public/chrome/ui/header/collapsible_nav_groups_label.tsx
Comment (SuZhou-Joe) [reply src/core/public/chrome/ui/header/collapsible_nav_groups_label.tsx:42]: I just tried with useLocalStorage and it seems we can not because the hook update the localStorage async, making the component react to the state incorrectly.

File: src/core/public/chrome/nav_group/nav_group_service.ts
Comment (SuZhou-Joe) [reply src/core/public/chrome/nav_group/nav_group_service.ts:186]: It can be but those nav links should have different categories and it is supposed that only the special features within a use case should be declared as `showInAllNavGroup`. Let's bear with it.

File: src/core/public/chrome/nav_group/nav_group_service.ts
Comment (SuZhou-Joe) [reply src/core/public/chrome/nav_group/nav_group_service.ts]: Sure, updated.

**Comment by SuZhou-Joe** [comment]
> do we need to update here to make sure nav group will not change when in all use case

Yes we should, updated.

**Review by ruanyl — DISMISSED** [review]

**Review by Hailong-am — APPROVED** [review]

File: src/core/public/chrome/nav_group/nav_group_service.ts
Comment (AMoo-Miki) [inline src/core/public/chrome/nav_group/nav_group_service.ts:143]: `total` can confuse developers who work on this as it indirectly means a number. The norm for naming this is `accumulator` or `acc` (not as good).

File: src/core/public/chrome/nav_group/nav_group_service.ts
Comment (AMoo-Miki) [inline src/core/public/chrome/nav_group/nav_group_service.ts:157]: Is there anyway we can reduce the amount of loops we run here?

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled.tsx
Comment (AMoo-Miki) [inline src/core/public/chrome/ui/header/collapsible_nav_group_enabled.tsx:65]: For later: Not a fan of having layout decisions made in JS. We should plan to move these to CSS, as classes that we toggle, or as CSS or SASS variables.

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx
Comment (AMoo-Miki) [inline src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx:108]: `currentNavGroup?.type` was true so `currentNavGroup` exists. `currentNavGroup?.title` doesn't need the `?`.

File: src/plugins/management/public/components/feature_cards/feature_cards.tsx
Comment (AMoo-Miki) [inline src/plugins/management/public/components/feature_cards/feature_cards.tsx:43]: Why risk it? It would be so much cheaper to just make a Map based on `link.category?.id` and group them than check the length of an array so many times.

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx
Comment (SuZhou-Joe) [reply src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx]: Thanks, updated.

File: src/plugins/management/public/components/feature_cards/feature_cards.tsx
Comment (SuZhou-Joe) [reply src/plugins/management/public/components/feature_cards/feature_cards.tsx]: Thanks, updated. Actually we do not need to get last group in every loop.

File: src/core/public/chrome/nav_group/nav_group_service.ts
Comment (SuZhou-Joe) [reply src/core/public/chrome/nav_group/nav_group_service.ts]: Sure, merged 2 loops into 1.

**Review by Hailong-am — APPROVED** [review]

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss
Comment (ruanyl) [inline src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss:5]: image
We should perhaps add `overflow: hidden` here, otherwise it looks like this

Comment (ruanyl) [inline src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss:5]: This background color doesn't look like the color in design, shall we use the color from design?

File: src/plugins/workspace/public/components/workspace_selector/workspace_selector.tsx
Comment (ruanyl) [inline src/plugins/workspace/public/components/workspace_selector/workspace_selector.tsx:1]: image

The workspace selector button color doesn't look right, I guess on line 91, it should be,

```typescript
      <EuiPanel paddingSize="s" borderRadius="m" color="transparent" hasBorder>
```

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss
Comment (ruanyl) [inline src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss:9]: Couldn't see the current selected app been highlighted in left nav, the reason is the left nav background color is the same as highlight nav item background color

image

File: src/core/public/chrome/ui/header/collapsible_nav_groups.tsx
Comment (ruanyl) [inline src/core/public/chrome/ui/header/collapsible_nav_groups.tsx:119]: `'data-test-subj': undefined` by intention?

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss
Comment (SuZhou-Joe) [reply src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss:5]: There is not a token for left nav in OUI yet, I'd like to merge this version first. We can raise another PR to adjust background color if we have any new updates.

Comment (ruanyl) [reply src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss:5]: It looks the background is one of the important change and affects the look & feel quite a bit, my comments below are also related, let me double check and ask UX to confirm.

If we'd like to merge this PR first, perhaps I'd rather prefer to use the color value from design first, then then replace it with the color defined in OUI color palettes when it's ready.

Comment (ruanyl) [reply src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss:5]: Or maybe we can have the background color change in a separate PR so that we can get the other changes in this PR merged asps.

Comment (ruanyl) [reply src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss:5]: @virajsanghvi has a PR to add the color to OUI https://github.com/opensearch-project/oui/pull/1430/files#diff-67cb04d06a675c67602ff65a876373033b16a759964be1b850bbfb4ca50bad6aR13

And confirmed with him that we can commit the changes and let him know so that he can update the color after the PR in OUI merged. @SuZhou-Joe

**Review by ruanyl — DISMISSED** [review]

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss
Comment (SuZhou-Joe) [reply src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss:5]: Sure, makes sense. I will use the color from design in this PR to unblock review.

File: src/core/public/chrome/nav_group/nav_group_service.ts
Comment (SuZhou-Joe) [reply src/core/public/chrome/nav_group/nav_group_service.ts]: Sure, updated.

File: src/core/public/chrome/nav_group/nav_group_service.ts
Comment (SuZhou-Joe) [reply src/core/public/chrome/nav_group/nav_group_service.ts:146]: Sure, updated.

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss
Comment (SuZhou-Joe) [reply src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss]: Thanks, fixed.

File: src/plugins/workspace/public/components/workspace_selector/workspace_selector.tsx
Comment (SuZhou-Joe) [reply src/plugins/workspace/public/components/workspace_selector/workspace_selector.tsx:1]: Sure, updated.

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss
Comment (SuZhou-Joe) [reply src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss:12]: Sure, by using the new background color value, it should be able to tell the difference.

File: src/core/public/chrome/ui/header/collapsible_nav_groups.tsx
Comment (SuZhou-Joe) [reply src/core/public/chrome/ui/header/collapsible_nav_groups.tsx:122]: Yes, added some comments.

**Review by Hailong-am — APPROVED** [review]

**Review by ruanyl — APPROVED** [review]

File: src/core/public/chrome/ui/header/variables.scss
Comment (Hailong-am) [inline src/core/public/chrome/ui/header/variables.scss:1]: should this go with theme change? when i switch to theme v7, the page looks like this

image

Comment (SuZhou-Joe) [reply src/core/public/chrome/ui/header/variables.scss:1]: It is. a intermedia state, will change to a variable in OUI theme after https://github.com/opensearch-project/oui/pull/1430/files#diff-67cb04d06a675c67602ff65a876373033b16a759964be1b850bbfb4ca50bad6aR13 get merged.

**Pull request merged**
